src/main/java/com/cityelf/service/UserService.java

```diff
@@ -46,23 +59,79 @@ public User getUser(long id) throws UserNotFoundException {
   }
 
 
-  public User addNewUser(User user)
-      throws UserException {
-    String firebaseId = user.getFirebaseId();
-    if (firebaseId == null) {
-      throw new UserNoFirebaseIdException();
+  public Statuses addNewUser(String firebaseId, String address)
+      throws UserAlreadyExistsException {
+    User newUser;
+
+    if (userRepository.findByFirebaseId(firebaseId) == null) {
+      userRepository.save(new User(firebaseId));
+      newUser = userRepository.findByFirebaseId(firebaseId);
+      Address address1 = addressesRepository.findByAddress(address);
+      long idAddress = address1.getId();
+      userAddressesRepository
+          .save(new UserAddresses(newUser.getId(), idAddress));
+      userRoleRepository.save(new UserRole(newUser.getId(), 1));
+      return Statuses.USER_ADD_IN_DB_OK;
     } else {
-      if (userRepository.findByFirebaseId(firebaseId) != null) {
-        throw new UserAlreadyExistsException();
+      return Statuses.USER_EXIIST;
+    }
+  }
+
+  public Statuses registration(String fireBaseID, String email, String password) {
+    User newUser = userRepository.findByFirebaseId(fireBaseID);
+
+    if (newUser != null && !fireBaseID.equals("WEB")) {
+
+      newUser.setEmail(email);
+      newUser.setPassword(password);
+      userRepository.save(newUser);
+      String msg = "http://localhost:8088/services/users/confirmregistration?id=" + newUser.getId()
+          + "&email=" + email;
+      mailSenderService.sendMail(email, "Confirm registration CityELF", msg);
+
+      return Statuses.USER_REGISTRATION_OK;
+    } else {
```

Comment:
The **else** is redundant here since you have **return** in **if**. The code will look cleaner without **else**.